src/Select.js

```diff
@@ -901,7 +901,8 @@ const Select = React.createClass({
 	},
 
 	renderClear () {
-		if (!this.props.clearable || (!this.props.value || this.props.value === 0) || (this.props.multi && !this.props.value.length) || this.props.disabled || this.props.isLoading) return;
+
+		if (!this.props.clearable || this.props.value == null || this.props.multi && !this.props.value.length || this.props.disabled || this.props.isLoading) return;
```

Comment:
I'm not keen on the loose equality operator - I know how it works, but given the number of people who use / contribute to this project, without an explicit comment it could be misinterpreted.

How about we replace this with a more explicit `value === undefined || value === null` so that the intended behaviour is explicit?